clarity/src/vm/tests/clarity_wasm/mod.rs

```diff
@@ -0,0 +1,36 @@
+use std::collections::{BTreeMap, BTreeSet};
+
+use ouroboros::self_referencing;
+use stacks_common::types::StacksEpochId;
+
+use crate::vm::analysis::{AnalysisDatabase, CheckResult, ContractAnalysis};
+use crate::vm::database::MemoryBackingStore;
+use crate::vm::errors::CheckErrors;
+use crate::vm::types::{
+    FunctionSignature, FunctionType, QualifiedContractIdentifier, TraitIdentifier,
+};
+use crate::vm::{ClarityName, ClarityVersion};
+
+#[self_referencing]
+pub struct MemoryAnalysisDatabase {
+    mem_store: MemoryBackingStore,
+    #[not_covariant]
+    #[borrows(mut mem_store)]
+    pub analysis_db: AnalysisDatabase<'this>,
+}
```

Comment:
Can we have some documentation and an explanation for why we actually need this?

Reply:
Implementation has changed.